src/main/kotlin/se/uulm/snowballr/backend/service/UserService.kt

```diff
@@ -148,4 +161,27 @@ class UserService(
 
         return tokens
     }
+
+    override suspend fun updateUser(request: UserOuterClass.User.Update): UserOuterClass.User {
+        val currentUser = userRepo.getUserById(GrpcContext.getUserIdFromContext())
+
+        // When the email should be changed, then check whether a user with the given email already exists
+        if (request.mask.pathsList.contains("email") && userRepo.doesUserExistByEmail(request.user.email)) {
+            throw DuplicateEntityException.UserEmail(request.user.email)
+        }
+
+        // When the role should be changed, then check whether the requesting user a server admin
+        if (request.mask.pathsList.contains("role") && currentUser.role != UserRole.USER_ROLE_ADMIN) {
+            throw UnauthorizedException.Single.User(currentUser.id.toString(), request.user.id, AccessType.UPDATE)
+        }
```

Comment:
Meanwhile, this helper was added.
```suggestion
        if (request.mask.pathsList.contains("role")) {
            verifyServerAdminRole(currentUser) { UnauthorizedException.Single.User(it, request.user.id, AccessType.UPDATE) }
        }
```

Reply:
Yeah thank you for the hint, but this does not work directly, as this method only accept an `UnauthorizedException.All`

Reply:
Yes, we have to find a way to generalize this after you rebase.

Reply:
Ok, for now I changed the signature to accept all kinds of Unauthorized Exceptions.